src/plugins/intel_npu/src/plugin/src/metadata.cpp

```diff
@@ -0,0 +1,123 @@
+// Copyright (C) 2018-2024 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "metadata.hpp"
+
+#include <cstring>
+#include <sstream>
+
+#include "intel_npu/config/config.hpp"
+#include "intel_npu/utils/logger/logger.hpp"
+#include "openvino/core/version.hpp"
+
+namespace intel_npu {
+
+OpenvinoVersion::OpenvinoVersion(const std::string& version) {
+    this->version = version;
+    this->size = static_cast<uint32_t>(version.size());
+}
+
+void OpenvinoVersion::read(std::istream& stream) {
+    stream.read(reinterpret_cast<char*>(&size), sizeof(size));
+    stream.read(&version[0], size);
+}
+
+Metadata<METADATA_VERSION_1_0>::Metadata() : ovVersion{ov::get_openvino_version().buildNumber} {
+    version = METADATA_VERSION_1_0;
+}
+
+void Metadata<METADATA_VERSION_1_0>::read(std::istream& stream) {
+    ovVersion.read(stream);
+}
+
+uint32_t& OpenvinoVersion::get_size() {
```

Comment:
I need it as reference, [I gave the reason above](https://github.com/openvinotoolkit/openvino/pull/27159/#discussion_r1872716488).

Since it is used only for tests, maybe I could move that method to a `Protected` field and make a test class which inherits it.

Reply:
> I need it as reference

No you don't, there are safer ways to do this. For instance, this is a primitive data type, inside `Metadata::write` you could just store it in a temporary variable and pass that to the stream. Or maybe it works as is without a temporary variable, haven't checked. Or you could create a method inside `OpenvinoVersion` to update the stream.

> Since it is used only for tests

I've not noticed this call in the test file, maybe I've missed it.

Reply:
I added `OpenvinoVersion::write` method to avoid using this getter.